# Case Study on Testing of Hypothesis
## A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.
   ### ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
   ###  ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.
#### Help the company to carry out their study with the help of data provided.


In [144]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from scipy.stats import chi2_contingency 

In [145]:
df=pd.read_csv('Sales_add.csv')

In [146]:
df.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [147]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


# 1.Descriptive analysis on the dataset.

In [148]:
df.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

## Insights:
    1.The given dataset contains 22 rows and 4 columns.
    2.This dataset is not containing any null values.
    3The average sales before digital marketing add is 149239.95 and average sales after digital add is 231123.73.

#  ●The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
## One tailed Paired Samples T-Test.
## Hypothesis:
*  **Ho**:The sales before and after stepping into digital marketing are same
*  **Ha**:The sales after stepping into digital marketing is more.
*  $\alpha $: 0.05



In [169]:
sales_BDM=df['Sales_before_digital_add(in $)']
sales_ADM=df['Sales_After_digital_add(in $)']

In [173]:
tvalue, pvalue=stats.ttest_rel(sales_ADM,sales_BDM,alternative='greater')
print("The t value of means is: ", round(tvalue, 4))
print("The p value based on the t score is: ", round(pvalue,4))

The t value of means is:  12.0907
The p value based on the t score is:  0.0


In [172]:
stats.ttest_rel(sales_ADM,sales_BDM,alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

## From above T test,  0 is the p-value, 12.0907 is the statistical value.
## As the p-value=0.0< $\alpha$=0.05 and hence we reject the null Hypothesis
## ie. The sales after stepping into digital marketing is more.

# ●The company needs to check whether there is any dependency between the features “Region” and “Manager”.
## Chi Square test for Dependency
## Hypothesis:
*  **Ho** :The 2 features “Region” and “Manager” which are to be compared are independent.
*  **Ha** :The 2 features are dependent.
*  $\alpha$ : 0.05



In [152]:
df3=pd.crosstab(df.Region, df.Manager, margins=True,values=df['Sales_before_digital_add(in $)'],aggfunc=np.sum)
df3

Manager     Manager - A  Manager - B  Manager - C      All
Region                                                    
Region - A       594514       466851       420684  1482049
Region - B       570900       152167       330598  1053665
Region - C       147463       469477       130625   747565
All             1312877      1088495       881907  3283279

In [153]:
valueBDM = np.array([df3.iloc[0][0:3].values,
                  df3.iloc[1][0:3].values,
                  df3.iloc[2][0:3].values])
stat, p, dof, expected=chi2_contingency(valueBDM)

In [154]:
print(chi2_contingency(valueBDM)[0:3])
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
    print('Dependent (reject H0)')
else:
    print('Independent (H0 holds true)')

(474356.9740349385, 0.0, 4)
p value is 0.0
Dependent (reject H0)


In [155]:
df4=pd.crosstab(df.Region, df.Manager, margins=True,values=df['Sales_After_digital_add(in $)'],aggfunc=np.sum)
df4

Manager     Manager - A  Manager - B  Manager - C      All
Region                                                    
Region - A      1030437       656832       701262  2388531
Region - B       939851       231808       429436  1601095
Region - C       229336       643654       222106  1095096
All             2199624      1532294      1352804  5084722

In [156]:
valueADM = np.array([df4.iloc[0][0:3].values,
                  df4.iloc[1][0:3].values,
                  df4.iloc[2][0:3].values])
stat, p, dof, expected=chi2_contingency(valueADM)

In [157]:
print(chi2_contingency(valueADM)[0:3])
alpha = 0.05
print("p value is " + str(p))
if p <= alpha:
    print('Dependent (reject H0)')
else:
    print('Independent (H0 holds true)')

(671476.3637089065, 0.0, 4)
p value is 0.0
Dependent (reject H0)


# From above two cases of considering both with and wihout Digital marketing,  0 is the p-value for both cases ,474356.97 and 671476.36 is the statistical value respectively and 4 is the degree of freedom. 
## As the p-value=0.0<  𝛼 =0.05 and hence we reject the NULL Hypothesis.
## Ie. The 2 features “Region” and “Manager” which are to be compared are dependent.